src/mps/dmrg.jl

```diff
@@ -278,7 +278,10 @@ function dmrg(PH, psi0::MPS, sweeps::Sweeps; kwargs...)
         end
 
         energy = vals[1]
-        phi::ITensor = vecs[1]
+        ## Right now there is a conversion problem in CUDA.jl where `UnifiedMemory` Arrays are being converted 
+        ## into `DeviceMemory`. This conversion line is here temporarily to fix that problem when it arises
+        phi::ITensor = itensor(convert(typeof(tensor(phi)), tensor(vecs[1])))
```

Comment:
An issue with this line is that in principle the type of the storage doesn't have to be exactly the same before and after call `eigsolve`, for example the input could have `Diag` storage but end up with `Dense` storage because of operations performed in `eigsolve`.

Could we use `adapt` here, like `adapt(leaf_parenttype(phi), vec[1])`? That would be more generic and also makes it clearer what it is doing.

Also maybe good to wrap in `iscu(phi) && iscu(vec[1])`, since also it's possible operations in `eigsolve` move the tensor back to CPU (intentionally), say if we have some feature in the future that moves tensors to CPU if GPU memory runs out.